hazelcast/src/test/java/com/hazelcast/internal/cluster/impl/MembershipUpdateTest.java

```diff
@@ -819,6 +827,79 @@ public void testListenerRegistrationWhileANewMemberJoining() {
         assertOpenEventually(entryAddedLatch);
     }
 
+    @Test
+    public void testDistributedObjectsAccurateAfterBatchedJoinHandling() throws InterruptedException {
+        // Prepare config
+        List<String> membersList = new ArrayList<>(10);
+        for (int k = 0; k < 10; k++) {
+            membersList.add("127.0.0.1:" + (5701 + k));
+        }
+        Config config = smallInstanceConfig();
+        config.setClusterName("myCluster").getNetworkConfig().getJoin().getTcpIpConfig().setMembers(membersList);
+
+        // Start our master instance
+        HazelcastInstance hz1 = Hazelcast.newHazelcastInstance(config);
+        // Start 9 members simultaneously (so their join request is batched)
+        HazelcastInstance[] members = new HazelcastInstance[9];
+        ExecutorService pool = Executors.newFixedThreadPool(9);
+        for (int k = 0; k < 9; k++) {
+            int finalK = k;
+            pool.execute(() -> members[finalK] = Hazelcast.newHazelcastInstance(config));
+        }
```

Comment:
Worth noting that here I am intentionally using non-test instances of Hazelcast - my tests were not able to reproduce this issue using the test variant.

Reply:
This is ok. However, you should change HazelcastParallelClassRunner -> *Serial* and remove `ParallelJVMTest`, otherwise your tests may influence other tests.

Reply:
Good catch, thank you @frant-hartm - I've done this, but the test now runs for ~100s instead of ~17s; do you think it's worth me moving my test out into its own class so we don't have to slow down all the others?

Reply:
Yes, every minute counts, especially for the tests which are not part of the `ParallelJVMTest` group.

Reply:
Done 👍 https://github.com/hazelcast/hazelcast/pull/25114/commits/4d08ded77afa549c4812e0582dd9632b4b8f44e5